sources/web/datalab/static/datalab.js

```diff
@@ -277,6 +277,168 @@ function initializePage(dialog, saveFn) {
+  if (!(CELL_METADATA_COLLAPSED in cell.metadata)) {
+    return;
+  }
+  let isHidden = typeof(hide) !== 'undefined' ? !hide : cell.metadata[CELL_METADATA_COLLAPSED];
+  let inputDiv = cell.cell_type === 'code' ? 'div.input' : 'div.inner_cell';
+  collapseSpan = cell.element.find('span.collapse-cell')[0];
+  if (isHidden) {
+    cell.element.find(inputDiv).show();
+    // for code cells, also show the output div
+    if (cell.cell_type == 'code')
+      cell.element.find('div.output_wrapper').show();
+    cell.element.find('div.cellPlaceholder').hide();
+    collapseSpan.classList.add(UNCOLLAPSE_BUTTON_CLASS);
+    collapseSpan.classList.remove(COLLAPSE_BUTTON_CLASS);
+  } else {
+    cell.element.find(inputDiv).hide();
+    // for code cells, also hide the output div
+    if (cell.cell_type == 'code')
+      cell.element.find('div.output_wrapper').hide();
+    cell.element.find('div.cellPlaceholder').show();
+    collapseSpan.classList.remove(UNCOLLAPSE_BUTTON_CLASS);
+    collapseSpan.classList.add(COLLAPSE_BUTTON_CLASS);
+  }
+  cell.metadata[CELL_METADATA_COLLAPSED] = !isHidden;
+}
+
+/**
+ * Collapse output of a cell
+ */
+function toggleCollapseCellOutput(cell, hide) {
+  if (!(CELL_METADATA_OUTPUT_COLLAPSED in cell.metadata)) {
+    return;
+  }
+  let isHidden = typeof(hide) !== 'undefined' ? !hide : cell.metadata[CELL_METADATA_OUTPUT_COLLAPSED];
+  collapseSpan = cell.element.find('span.collapse-output')[0];
+  if (isHidden) {
+    cell.expand_output();
+    collapseSpan.classList.add(COLLAPSE_OUTPUT_BUTTON_CLASS);
+    collapseSpan.classList.remove(UNCOLLAPSE_OUTPUT_BUTTON_CLASS);
+  } else {
+    cell.collapse_output();
+    collapseSpan.classList.remove(COLLAPSE_OUTPUT_BUTTON_CLASS);
+    collapseSpan.classList.add(UNCOLLAPSE_OUTPUT_BUTTON_CLASS);
+  }
+  cell.metadata[CELL_METADATA_OUTPUT_COLLAPSED] = !isHidden;
+}
+
+/**
+ * Create an HTML button for the cell minitoolbar and return it
+ */
+function createCellMiniToolbarButton(classNames, title, callback) {
+  let buttonDiv = document.createElement('button');
+  buttonDiv.className = 'btn btn-default';
+  buttonDiv.innerHTML = '<span class="' + classNames + '"></span>';
+  buttonDiv.title = title;
+  buttonDiv.addEventListener('click', callback);
+  return buttonDiv;
+}
+
+/**
+ * Patch the cell's element to add the hoverable minitoolbar
+ */
+function addCellMiniToolbar(cell) {
+
+  let hoverableDiv = document.createElement('div');
+  hoverableDiv.className = 'hoverable';
+
+  // only for code cells, add collapse-output and clear buttons
+  if (cell.cell_type === 'code') {
+    // collapse output button
+    let hideButton = createCellMiniToolbarButton(
+      'collapse-output fa ' + COLLAPSE_OUTPUT_BUTTON_CLASS,
+      'Collapse/Expand output',
+      function() {
+        toggleCollapseCellOutput(cell);
+      }
+    );
+    cell.element.find('div.output_collapsed')[0].addEventListener('click', function() {
+      toggleCollapseCellOutput(cell, false /*hide*/);
```

Comment:
> false /\*hide\*/

This was initially confusing to me, as the variable is named 'hide', and we are setting that to false, so this means 'do not hide'. However, the comment just says 'hide', which (at first) reads as meaning the exact opposite of what it does.

Can we maybe split "expandCellOutput" and "collapseCellOutput" helper methods out of the current body of "toggleCollapseCellOutput", and then just directly call "expandCellOutput"?